test/test_location.js

```diff
@@ -189,6 +200,20 @@ describe('states.location', function() {
                 .run();
         });
 
+        it('should go back to initial page if retry is selected',
+        function() {
+            tester.data.opts.retry_text = 'Retry';
+            return tester
+                .inputs("Friend Street", 'n', 'n', 'n', 'n', 'n', 'n', 'n', 'n',
+                    'n', 'n', 'n', 'n', 'n', 'n', 'n', 'r')
```

Comment:
We can maybe skip some / all of the `'n'`s after my comment about generate footer has been addressed.